docs/proposals/DNSPolicy.md

```diff
@@ -0,0 +1,409 @@
+      recordTTL: 60
+      recordType: A
+      setIdentifier: 172.32.0.2
+      providerSpecific:
+        - name: "aws/weight"
+          value: "60"
+      labels:
+        id: 172.32.0.2
+      targets:
+        - 172.32.0.2        
+    - dnsName: dnsrecord-geo.ie.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      setIdentifier: 172.32.0.3
+      providerSpecific:
+        - name: "aws/weight"
+          value: "60"
+      labels:
+        id: 172.32.0.3
+      targets:
+        - 172.32.0.3              
+
+...
+
+```
+
+
+
+### Weighted
+
+Weighted is effectively the same a GEO weighted but without the GEO records. Below is an evenly or Round Robin weighted record set. Below is an example record for Google DNS 
+
+
+```yaml 
+
+    - dnsName: weighted.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      setIdentifier: 172.32.0.2
+      providerSpecific:
+        - name: "google/weight"
+          value: "60"
+      labels:
+        id: 172.32.0.2
+      targets:
+        - 172.32.0.2        
+    - dnsName: weighted.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      setIdentifier: 172.32.0.3
+      providerSpecific:
+        - name: "google/weight"
+          value: "60"
+      labels:
+        id: 172.32.0.3
+      targets:
+        - 172.32.0.3              
+
+```
+
+
+### Custom Weighting 
+
+Custom weighting will use the associated attribute set on the `ManagedCluster` to decide which records should get a specific weight. 
+
+example:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+ annotations:
+   kuadrant.io/lb-attribute-cloud: "GCP"
+```
+
+Is associated with a DNSPolicy where `kuadrant.io/lb-attribute-` is the prefix.
+
+```YAML
+  - attribute: cloud
+    value: GCP
+    weight: 20
+```        
+
+So any gateway placed on this `ManagedCluster` targeted by this DNSPolicy will get an A record with a weight of 20 
+
+### Conflict Resolution
+
+As a cluster can have multiple attributes it is possible that a DNSPolicy will define two weights for a single cluster. If a DNSPolicy defines two different weights  
+
+Example:
+```yaml 
+
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+ annotations:
+   kuadrant.io/lb-attribute-cloud: "AWS"
+   kuadrant.io/lb-attribute-compute: "GC"
+---
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+spec:
+  weighted:
+   default: 10
+  custom:
+  - attribute: cloud 
+    value: AWS
+    weight: 20
+  - attribute: compute
+    value: GC
+    weight: 30
+```
+
+With the above defined managed cluster, this weighting would give us two values. 10 and 20. Proposal is in this case is that we cannot pick any weight and so the DNSPolicy is invalid and a status condition should be set. In this case we would not change any DNS records.
+An alternative in this case would be to pick the default weight, but update the status to ensure it is clear this has been due to the conflict.
```

Comment:
I think the alternative is a better option personally, or we use the the smallest weight value of all the possible ones.

Reply:
I agree with Mike on this. I think making it possible for DNS Records to enter an "invalid" state reduces our ability to perform one of our main goals (pointing DNS Records at the correct clusters).

Reply:
Yes I think we go with, you have to set a default and if the policy is invalid / conflicted we fall back to the default and ensure it is clear what has happened in the status. 
Perhaps in a future iteration we handle this via a validating webhook?

Reply:
An alternative spec to avoid conflict would be to limit the attribute to 1 fixed name.
e.g. `kuadrant.io/lb-attribute-weight`
In the DNSPolicy, the value is the sole differentiator e.g.
```yaml
  - value: cloud-aws
    weight: 20
  - value: compute-gc
    weight: 30
```
It much more restrictive, and forces the user to think about the attribute more, but does avoid conflicts.

Reply:
I think for the usecases we have the single attribute is enough. It allows different values on different clusters and allowed different weighting for different gateways

Reply:
it simplifies the implementation and is easier  harder to get wrong from an end user perspective

Reply:
```kuadrant.io/lb-attribute-custom-weight: "GCP"```